allennlp_models/vision/dataset_readers/vgqa.py

```diff
@@ -0,0 +1,258 @@
+            try:
+                processed_images = self._process_image_paths(
+                    self.images[filename] for filename in filenames
+                )
+            except KeyError as e:
+                print(self.images)
+                missing_id = e.args[0]
+                raise KeyError(
+                    missing_id,
+                    f"We could not find an image with the id {missing_id}. "
+                    "Because of the size of the image datasets, we don't download them automatically. "
+                    "Please go to https://visualqa.org/download.html, download the datasets you need, "
+                    "and set the image_dir parameter to point to your download location. This dataset "
+                    "reader does not care about the exact directory structure. It finds the images "
+                    "wherever they are.",
+                )
+        else:
+            processed_images = [None for _ in range(len(question_dicts))]
+
+        logger.info("Reading the dataset")
+        failed_instances_count = 0
+        attempted_instances_count = 0
+        for qa, processed_image in zip(question_dicts, processed_images):
+            question = qa["question"]
+            answer = utils.preprocess_answer(qa["answer"])
+            qa_id = qa["qa_id"]
+
+            instance = self.text_to_instance(
+                qa_id,
+                question,
+                answer,
+                processed_image,
+            )
+
+            attempted_instances_count += 1
+            if instance is not None:
+                yield instance
+            else:
+                failed_instances_count += 1
+
+        failed_instances_fraction = failed_instances_count / attempted_instances_count
+        logger.warning(f"{failed_instances_fraction*100:.0f}% of instances failed.")
+
+    @overrides
+    def text_to_instance(
+        self,  # type: ignore
+        qa_id: int,
+        question: str,
+        answer: str,
+        image: Union[str, Tuple[Tensor, Tensor]],
+        use_cache: bool = True,
+        keep_impossible_questions: bool = True,
+    ) -> Optional[Instance]:
+        question_field = TextField(self._tokenizer.tokenize(question), None)
+
+        fields: Dict[str, Field] = {
+            "question": question_field,
+        }
+
+        if image is None:
+            return None
+
+        if image is not None:
+            if isinstance(image, str):
+                features, coords = next(self._process_image_paths([image], use_cache=use_cache))
+            else:
+                features, coords = image
+
+            fields["box_features"] = ArrayField(features)
+            fields["box_coordinates"] = ArrayField(coords)
+            fields["box_mask"] = ArrayField(
+                features.new_ones((features.shape[0],), dtype=torch.bool),
+                padding_value=False,
+                dtype=torch.bool,
+            )
+
+        if answer is not None:
```

Comment:
`answer` needs that `Optional[str]` annotation. `answer` is not given when we are running predictions only.